modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/side_navigation.es.js

```diff
@@ -427,6 +447,206 @@ SideNavigation.prototype = {
+						content.getBoundingClientRect().right;
+
+					let paddingRight = options.gutter + options.width;
+
+					const contentMaxWidth =
+						getComputedStyle(content).maxWidth ||
+						getComputedStyle(content).width;
+
+					if (contentMaxWidth.match(/px$/g)) {
+						contentMargin =
+							(document.body.scrollWidth -
+								toInt(contentMaxWidth)) /
+							2;
+
+						if (contentMargin > options.width) {
+							paddingRight = '';
+						}
+						else if (
+							contentMargin > 0 &&
+							contentMargin < options.width
+						) {
+							paddingRight =
+								options.gutter + options.width - contentMargin;
+						}
+					}
+
+					addClass(container, 'sidenav-transition');
+
+					setStyles(content, {
+						'padding-right': px(paddingRight),
+					});
+				}
+			};
+
+			document.addEventListener(
+				'closedStart.lexicon.sidenav',
+				instance._handleClosedStart
+			);
+		}
+	},
+
+	_onOpen() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+
+		if (!instance._handleOpen) {
+			instance._handleOpen = () => {
+				const type = !instance._isDesktop()
+					? options.typeMobile
+					: options.type;
+
+				if (type === 'relative' && hasClass(container, 'open')) {
+					removeClass(container, 'sidenav-transition');
+				}
+			};
+
+			document.addEventListener(
+				'open.lexicon.sidenav',
+				instance._handleOpen
+			);
+		}
+	},
+
+	_onOpenStart() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+		const content = document.querySelector(options.content);
+
+		if (!instance._handleOpenStart) {
+			instance._handleOpenStart = (event) => {
+				const type = !instance._isDesktop()
+					? options.typeMobile
+					: options.type;
+
+				if (
+					type === 'relative' &&
+					hasClass(container, 'open') &&
+					content.closest('.page-maximized')
+				) {
+					const otherMenu = document.querySelector(
+						event.detail.options.container + ' .sidenav-menu'
+					);
+					const otherMenuWidth = otherMenu.getBoundingClientRect()
+						.width;
+
+					const contentMargin =
+						document.body.scrollWidth -
+						content.getBoundingClientRect().right -
+						otherMenuWidth / 2;
+
+					const paddingRight =
+						contentMargin > 0
+							? options.width + options.gutter - contentMargin
+							: options.width + options.gutter;
+
+					addClass(container, 'sidenav-transition');
+
+					setStyles(content, {
+						'padding-right': px(paddingRight),
+					});
+				}
+			};
+
+			document.addEventListener(
+				'openStart.lexicon.sidenav',
+				instance._handleOpenStart
+			);
+		}
+	},
+
+	_onScreenChange() {
+		const instance = this;
+		const options = instance.options;
+
+		const container = document.querySelector(options.container);
+		const content = document.querySelector(options.content);
+
+		let startRegion = instance._isDesktop();
```

Comment:
Is `region` the right word for this? Alternatives would be just `isDesktop`, `originalIsDesktop`, `startingIsDesktop`, `prevousIsDesktop` etc.